nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -22,33 +27,43 @@ let
   wrapped = runCommand "ipfs" { buildInputs = [ makeWrapper ]; preferLocalBuild = true; } ''
     mkdir -p "$out/bin"
     makeWrapper "${ipfs}/bin/ipfs" "$out/bin/ipfs" \
-      --set IPFS_PATH ${cfg.dataDir} \
+      --set IPFS_PATH "${cfg.dataDir}" \
       --prefix PATH : /run/wrappers/bin
   '';
 
 
   commonEnv = {
     environment.IPFS_PATH = cfg.dataDir;
-    path = [ wrapped ];
+    path = [ wrapped utillinux ];
     serviceConfig.User = cfg.user;
     serviceConfig.Group = cfg.group;
   };
 
   baseService = recursiveUpdate commonEnv {
     wants = [ "ipfs-init.service" ];
     # NB: migration must be performed prior to pre-start, else we get the failure message!
-    preStart = optionalString cfg.autoMount ''
+    preStart = ''
+      install -m 0755 -o "${cfg.user}" -g "${cfg.group}" -d "${cfg.dataDir}"
```

Comment:
is this still needed in ipfs 0.5.1? ideally we would get some more diagnostics and report this as a bug